src/core/attributeformmodelbase.cpp

```diff
@@ -808,7 +808,7 @@ void AttributeFormModelBase::updateVisibilityAndConstraints( int fieldIndex )
     if ( fieldIndex != -1 && fidx != fieldIndex )
     {
       const QString fieldName = mLayer->fields().at( fieldIndex ).name();
-      const QgsExpression expression = mLayer->fields().at( fieldIndex ).constraints().constraintExpression();
+      const QgsExpression expression = mLayer->fields().at( fidx ).constraints().constraintExpression();
```

Comment:
I believe there's a problem that it's not clear what `fieldIndex` is and what `fidx` is. Can we clarify that by either renaming and/or adding comments?